test/types/string/validation/nonUTFEscape.chpl

```diff
@@ -0,0 +1,32 @@
+var myBytes = b"abc\xffa";
+
+writeln("Base bytes: ");
+for b in myBytes.bytes() {
+  writeln(b);
+}
+writeln();
+
+var myString = myBytes.decode(errors=decodePolicy.escape);
+writeln("Escaped string bytes: ");
+for b in myString.bytes() {
+  writef("Decimal: %u, Hexadecimal: %xu\n", b, b);
+}
+writeln();
+
+writeln("Direct copy string bytes: ");
+var directBytes = myString.encode(errors=encodePolicy.pass);
+for b in directBytes.bytes() {
+  writeln(b);
+}
+writeln();
+
+writeln("Unescaped string bytes: ");
+var unescapedBytes = myString.encode(errors=encodePolicy.unescape);
+for b in unescapedBytes.bytes() {
+  writeln(b);
+}
+writeln();
+
+writeln("Should be false: ", myBytes == directBytes);
+writeln("Should be true: ", myBytes == unescapedBytes);
```

Comment:
Can you add some more testing? I'd like to see some testing that creates random `bytes` strings and encodes them with escaping and then decodes them unescaping and makes sure that they match.

I'd also like to see some tests of the replacement character policy that we're adding the right number of replacement characters according to Unicode standards / best practices. Feel free to make that a TODO for yourself to handle in another PR.